src/string/builder.cr

```diff
@@ -61,6 +61,29 @@ class String::Builder
     @bytesize == 0
   end
 
+  # Chomps the last byte from the string buffer.
+  # If the byte is '\n' and there's a '\r' before it, it is
+  # also removed.
+  def chomp!(byte : UInt8)
+    if bytesize > 0 && buffer[bytesize - 1] == byte
+      back(1)
+
+      if byte === '\n' && bytesize > 0 && buffer[bytesize - 1] === '\r'
```

Comment:
For me it's OK. I'll merge this, please send a PR to fix this if you find a way.

The problem is, when you do `gets`, this is equivalent to doing `gets('\n')`, so passing no delimiter is equivalent to passing `\n` as a delimiter. If `\n` is the delimiter, and the line ends with `\r\n`, then `\r\n` is also removed. There's currently no way to distinguish between "no delimiter is specified" and "`\n` was specified as a delimiter". To do this, we'd have to change the general `gets(delimiter, limit, chomp)` method to be something like `gets(delimiter, limit, chomp, argless)`, or do a separate implementation for the argless case in several `IO` types.

I personally don't find a use case where you'd want to remove `\n` but keep the `\r`, but if you do find such case, send a PR with all the necessary changes.